### PR TITLE
Add ECS logging MVP to libbeat

### PR DESCRIPTION
**Pull request by simitt**
## What does this PR do?
Enable ECS conformant logging by setting `logging.ecs:true`
Adds @timestamp, log.level, message and ecs.version to every log line.
If EnableCaller is set add log.origin.file and log.origin.line.
Logs error.message when error is given

## Why is it important?
Get started with ECS conformant logging

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
Set different logging options (syslog, to_files, console) and enable `logging.json` and `logging.ecs` and check log lines contain `@timestamp, message, log.level, ecs.version`. Logged errors are logged in the format `{ "error": { "message": .. } }`. If caller is enabled it is logged as `log.origin.*`. 

manually tested for 
* `logging.to_files: true`
* `logging.to_syslog: true`
* `logging.to_stderr: true`
* log output to console `./beat -e`

## Related issues

implements elastic/beats#17942

## Logs
Example:
```
{"log.level":"error","@timestamp":"2020-04-24T16:03:13.921+0200","log.origin":{"file.name":"instance/metrics.go","file.line":92},"message":"Error while getting memory usage: error retrieving process stats: cannot find matching process for pid=13762","ecs.version":"1.5.0"}
```

### CODE REVIEW

File: libbeat/logp/config.go [hunk 1]
```diff
@@ -24,10 +24,11 @@ import (
...
 type Config struct {
-	Beat      string   `config:",ignore"`   // Name of the Beat (for default file name).
-	JSON      bool     `config:"json"`      // Write logs as JSON.
-	Level     Level    `config:"level"`     // Logging level (error, warning, info, debug).
-	Selectors []string `config:"selectors"` // Selectors for debug level logging.
+	Beat       string   `config:",ignore"`   // Name of the Beat (for default file name).
+	JSON       bool     `config:"json"`      // Write logs as JSON.
+	Level      Level    `config:"level"`     // Logging level (error, warning, info, debug).
+	Selectors  []string `config:"selectors"` // Selectors for debug level logging.
+	ECSEnabled bool     `config:"ecs"`
```
Comment (urso) [inline libbeat/logp/config.go:31]: we want to enable this setting by default?

Comment (simitt) [reply libbeat/logp/config.go:31]: I would consider changing the default a breaking change, as it changes the log format.

**Comment by simitt** [comment]
Test failures are unrelated and also observed in beats `master` branch.

**Comment by simitt** [comment]
@urso I just updated to the latest master, would appreciate a review.

File: libbeat/logp/config.go (hunk 1 above)
Comment (kvch) [reply libbeat/logp/config.go:31]: Do you mind adding a comment to describe what this option does?

Comment (simitt) [reply libbeat/logp/config.go]: good point - added a comment

**Comment by urso** [comment]
The run on jenkins was green, but failed when reporting the results.

**Comment by simitt** [comment]
Thank you both for the review.

**Comment by simitt** [comment]
@urso, @kvch this PR was backported to `7.8`. The flag `logging.ecs` signals that ECS compliant logging is enabled, but this only introduces the [minimal required set of keys](https://github.com/elastic/ecs-logging/tree/master/spec#minimum-viable-product) for ECS logging. 
I think we should label the flag `experimental` for all beats until they actually adopted to ECS compliant logging. WDYT?

**Comment by urso** [comment]
Hm, when do you plan to make it the default? At least when running via Agent we should have ECS enabled by default.

**Comment by kvch** [comment]
Why experimental? If we mark the feature experimental, it implies (at least to me) that we might get rid of the feature or change it completely. But if I understand correctly this is the final format, we are just adding more fields in later releases.

However, I agree we should flag it somehow. So I would rather go with beta. Unless of course, the this functionality is subject to change.

**Comment by simitt** [comment]
My concern is that this adds a minimal set of ECS fields for now. It doesn't ensure that all log messages are ECS compatible, as this is still up to the implementer of the log messages. 

@kvch agreed, beta would be a better fit. I'd just like to avoid the assumption that as soon as you set `logging.ecs: true` all your logs are ECS aligned, as that's simply not the case yet.

**Comment by felixbarny** [comment]
> Hm, when do you plan to make it the default? At least when running via Agent we should have ECS enabled by default.

++

Is it the default already when running under agent? If not, is there a follow-up issue tracking this?

**Review by andrewvc — COMMENTED** [review]
LGTM for the minimal heartbeat changes

**Pull request merged**